src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndex.cs

```diff
@@ -46,6 +46,20 @@ public IFunctionDefinition Lookup(string functionId)
             return _functionsById[functionId];
         }
 
+        public IFunctionDefinition LookupByName(string name)
```

Comment:
lookup by which name type? Can we make it more explicit from the function name or definition?